libs/api/transaction/data-access/src/lib/api-transaction.service.ts

```diff
@@ -170,7 +170,7 @@ export class ApiTransactionService implements OnModuleInit {
       lastValidBlockHeight: input?.lastValidBlockHeight,
       mintPublicKey: mint?.mint?.address,
       processingStartedAt,
-      reference: input?.reference,
+      reference: input.reference || createReference(input.referenceType, input.referenceId),
```

Comment:
Should we keep the optional chaining `?`?

Reply:
`input` should always be set, but adding it to be sure!